core/src/Microsoft.Teams.BotApps/Schema/TeamsActivity.cs

```diff
@@ -0,0 +1,143 @@
+    [JsonConstructor]
+    public TeamsActivity()
+    {
+        From = new TeamsConversationAccount(new ConversationAccount());
+        Recipient = new TeamsConversationAccount(new ConversationAccount());
+        Conversation = new TeamsConversation(new Conversation());
+
+    }
+
+    private TeamsActivity(CoreActivity activity)
+    {
+        Id = activity.Id;
+        ServiceUrl = activity.ServiceUrl;
+        ChannelId = activity.ChannelId;
+        Type = activity.Type;
+        // ReplyToId = activity.ReplyToId;
+        Text = activity.Text;
+        Entities = EntityList.FromJsonArray(activity.Entities);
+        ChannelData = new TeamsChannelData(activity.ChannelData!);
+        From = new TeamsConversationAccount(activity.From!);
+        Recipient = new TeamsConversationAccount(activity.Recipient!);
+        Conversation = new TeamsConversation(activity.Conversation!);
+        Attachments = TeamsAttachment.FromJArray(activity.Attachments);
+
+        //base.Entities = Entities.ToJsonArray();
+        Rebase();
+    }
+
+    /// <summary>
+    /// resets shadow properties in base class
+    /// </summary>
+    /// <returns></returns>
+    internal TeamsActivity Rebase()
+    {
+        base.Entities = this.Entities?.ToJsonArray();
+        base.ChannelData = this.ChannelData;
+        base.From = this.From;
+        base.Recipient = this.Recipient;
+        base.Conversation = this.Conversation;
+
+        return this;
+    }
+
+    /// <summary>
+    /// Gets or sets the account information for the sender of the Teams conversation.
+    /// </summary>
+    [JsonPropertyName("from")] public new TeamsConversationAccount From { get; set; }
+
+    /// <summary>
+    /// Gets or sets the account information for the recipient of the Teams conversation.
+    /// </summary>
+    [JsonPropertyName("recipient")] public new TeamsConversationAccount Recipient { get; set; }
+
+    /// <summary>
+    /// Gets or sets the conversation information for the Teams conversation.
+    /// </summary>
+    [JsonPropertyName("conversation")] public new TeamsConversation Conversation { get; set; }
+
+    /// <summary>
+    /// Gets or sets the Teams-specific channel data associated with this activity.
+    /// </summary>
+    [JsonPropertyName("channelData")] public new TeamsChannelData? ChannelData { get; set; }
+
+    /// <summary>
+    /// Gets or sets the entities specific to Teams.
+    /// </summary>
+    [JsonPropertyName("entities")] public new EntityList? Entities { get; set; }
+
+    /// <summary>
+    /// Attachments specific to Teams.
+    /// </summary>
+    [JsonPropertyName("attachments")] public new IList<TeamsAttachment>? Attachments { get; set; }
+
+    /// <summary>
+    /// Adds an entity to the activity's Entities collection.
+    /// </summary>
+    /// <param name="entity"></param>
+    /// <returns></returns>
+    public TeamsActivity AddEntity(Entity entity)
+    {
+        Entities ??= [];
+        Entities.Add(entity);
```

Comment:
Pick up nuances from existing entities. For eg. there can only be 1 MessageEntity